### DISCUSSION TITLE
Reshape a CSV by pairing all fields but one named 'date'

### DISCUSSION THREAD

**Discussion by NikosAlexandris** [johnkerl/miller#658] [Show and tell]
I am recording this here as a future reference for me, and perhaps someone else. I've had a hard time to get a regular expression to match all field names but the one named `date`.

I have a series of fields like `date`, `i1`, `i2`, `j1`, `j2`, `k1`, `k2`. Then, I define PAIRS=`i1,i2,j1,j2,k1,k2` which I want to correlate like
```bash
mlr.go --csv cat input.csv |mlr.go reshape -i "$PAIRS" -o covariables,correlation
```
And this works fine.

Consider that the fields other than `date` are many more. Hence, I would like to get this running via a regular expression by matching all _other_ field names. Before finally understanding the simple syntax required, 
```bash
mlr.go reshape -r '[^date]' -o covariables,correlation
```

I was trying many things like
```bash
-r '!(^date)'
-r '!^date'
-r '!date'
-r '?!date'
-r '[^d]'
-r '[^d]*'
-r '[^da*]*'
-r '[^date*]'
-r '[^date]*'
-r '^((?!date).)*$'
-r '^((?!date).)*'
-r '^(?!date)(.*)'
-r '^(?!date).)*'
-r '^(?!date).*$'
-r '^(?!date).*'
-r '^?!date.*$'
-r '^[?!date].*'
-r '^\(\(?!date\).\)*$'
-r '^date'
-r '^date*'
-r [^date]*
-r ^date*
-r date*
```
😆

**Comment by aborruso** [comment]
If you asked right here, I had the answer ready for you🙃

**Reply by NikosAlexandris** [reply]
Learning process... :-)
